RoddyCore/src/de/dkfz/roddy/knowledge/files/BaseFile.groovy

```diff
@@ -93,16 +96,20 @@ public abstract class BaseFile<FS extends FileStageSettings> extends FileObject
         }
 
         public ConstructionHelperForGenericCreation(ExecutionContext context, ToolEntry creatingTool, String toolID, String slotID, String selectionTag, FileStageSettings fileStageSettings, JobResult jobResult) {
-            super(context, fileStageSettings, selectionTag, jobResult)
+            super(context, fileStageSettings, selectionTag, null, jobResult)
             this.slotID = slotID
             this.toolID = toolID
             this.creatingTool = creatingTool
         }
     }
 
     public static class ConstructionHelperForManualCreation extends ConstructionHelperForGenericCreation {
+        public ConstructionHelperForManualCreation(FileObject parentObject, List<FileObject> parentFiles, ToolEntry creatingTool, String toolID, String slotID, String selectionTag, String indexInFileGroup, FileStageSettings fileStageSettings, JobResult jobResult) {
```

Comment:
The thing is, this class is a helper object. It's sole purpose is to provide all necessary items to create an object. At least with Idea you get a very very good idea of what is happening, when you debug it (it prints you the var names and the contents). I don't know, if a map really helps you to make it better. We could however make a builder class for it and provide both.